### PR TITLE
AuthToken Plugin && BrowserStorage Plugin

### PR DESCRIPTION
**Pull request by richard-hulm**
Exposes two services:

- BrowserStorage: Exposes LocalStorage in an generic way, so we can swap out how things are stored if needed
- AuthToken: In-Memory store for auth tokens, so then on the application side, we can have an Authentication provider specific plugin, that just updates the in-memory store for an auth token, and then everything that needs it can read from this centralised place.

### CODE REVIEW

**Comment by Cowbacca** [comment]
Missing examples and tests but otherwise the code itself looks good.

File: examples/authentication/src/auth/demo-auth/DemoAuthPlugin.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import {PluginConfig, service} from '@brightinteractive/bright-js-framework'
+import TokenInBrowserStorageService from './TokenInBrowserStorageService'
+
+export const AUTH0_WEB_AUTH = 'auth0WebAuth'
```
Comment (Cowbacca) [inline examples/authentication/src/auth/demo-auth/DemoAuthPlugin.ts]: Should be deleted I guess?

Comment (kevinbennett) [reply examples/authentication/src/auth/demo-auth/DemoAuthPlugin.ts]: Yep. Done.

File: examples/authentication/src/auth/demo-auth/TokenInBrowserStorageService.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import {AuthTokenActions, Token} from '@brightinteractive/bright-js-framework/plugins/auth/auth'
+import {authTokenActions} from '@brightinteractive/bright-js-framework/plugins/auth'
+import {browserStorage, BrowserStorageSystem} from '@brightinteractive/bright-js-framework/plugins/browserstorage'
+import {Service} from '@brightinteractive/bright-js-framework'
+
+export default class TokenInBrowserStorageService extends Service<any> {
+
+    ID_TOKEN_KEY = 'ap_auth0_id_token'
```
Comment (Cowbacca) [inline examples/authentication/src/auth/demo-auth/TokenInBrowserStorageService.ts]: Probably change this to not have auth0 in the string

File: examples/authentication/src/pages/IndexPage.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  loginAsAdmin = () => {
+    this.demoAuthService.login('admin', 'password')
+  }
+
+  render() {
+    return (
+      <div>
+        <IsLoggedIn>
+          Log in succeeded!
```
Comment (Cowbacca) [inline examples/authentication/src/pages/IndexPage.tsx]: Should we add a logout button as well?

File: examples/authentication/src/auth/demo-auth/TokenInBrowserStorageService.ts (hunk 2 above)
Comment (kevinbennett) [reply examples/authentication/src/auth/demo-auth/TokenInBrowserStorageService.ts]: Done.

**Review by kevinbennett — APPROVED** [review]

**Pull request merged**